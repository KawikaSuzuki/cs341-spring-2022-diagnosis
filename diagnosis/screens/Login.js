import React, { useState } from "react";
import { View, Text, ActivityIndicator } from "react-native";
import { StatusBar } from "expo-status-bar";
import { StackActions } from "@react-navigation/native";
//formik
import { Formik } from "formik";

import {
  StyledContainer,
  InnerContainer,
  PageLogo,
  PageTitle,
  StyledForm,
  LabelIcon,
  Colors,
  StyleTextInput,
  EyeIcon,
  StyledButton,
  ButtonText,
  MessageBox,
  Line,
  ExtraView,
  ExtraText,
  TextLink,
  TextLinkContent,
  StyleInputLabel,
} from "../components/SignUpLoginStyles";

import { Octicons, Ionicons, Fontisto } from "@expo/vector-icons";
import axios from "axios";
//colors
const { company, placeholder, textInputBackground } = Colors;

//keyboard avoiding wrapper
import KeyboardAvoidWrap from "../components/KeyboardAvoidWrap";

const Login = ({ navigation }) => {
    const [hidePassword, setHidePassword] = useState(true);
    const [message, setMessage] = useState("");
    const [messageType, setMessageType] = useState("");

    const handleMessage = (message, type) => {
        setMessage(message);
        setMessageType(type);
    };
    return (
        <KeyboardAvoidWrap>
            <StyledContainer>
                <StatusBar style="dark" />
                <InnerContainer>
                    <PageLogo
                        resizeMode="contain"
                        source={require("./../assets/cropped-logo_new-1.png")}
                    />
                    <PageTitle>Account Login</PageTitle>
                    <Formik
                        initialValues={{ username: "", password: "" }}
                        onSubmit={(values, { setSubmitting }) => {
                            handleMessage(null);
                            if (values.username == "" || values.password == "") {
                                handleMessage("Please fill out all the fields above.");
                                setSubmitting(false);
                            } else {
                                fetch("https://up.physicaldiagnosispdx.com/up/app-content/authentication.php", {
                                    method: 'POST', 
                                    headers: {
                                        'Accept': 'application/json, text/plain, */*',
                                        'Content-Type': 'application/json'
                                    },
                                    body: JSON.stringify({username: values.username, password: values.password}) 
                                }).then(
                                    response => response.text()
                                ).then(function(data) {
                                    if(data === "Authenticated"){
                                        setSubmitting(false);
                                        navigation.navigate("Main Menu");
                                    }
                                    else{
                                        setSubmitting(false);
                                        handleMessage(data);
                                    }
                                }).catch(function(error) {  
                                    console.error(error);
                                    handleMessage("An error occurred. Check your network and try again.");
                                });
                            }
                        }}
                    >{({handleChange, handleBlur,handleSubmit,values,isSubmitting}) => (
                        <StyledForm>
                            <TextInput
                                label="Username"
                                icon="person"
                                placeholder="username"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange("username")}
                                onBlur={handleBlur("username")}
                                value={values.username}
                                testID="username-input"
                            />
                            <TextInput
                                label="Password"
                                icon="lock"
                                placeholder="********"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange("password")}
                                onBlur={handleBlur("password")}
                                value={values.password}
                                secureTextEntry={hidePassword}
                                isPassword={true}
                                hidePassword={hidePassword}
                                setHidePassword={setHidePassword}
                                testID="password-input"
                            />
                            <MessageBox testID="msgbox" type={messageType}>{message}</MessageBox>
                            {!isSubmitting && (
                                <StyledButton testID="loginbutton" onPress={handleSubmit}>
                                    <ButtonText>Login</ButtonText>
                                </StyledButton>
                            )}

                            {isSubmitting && (
                                <StyledButton testID="activitybutton" disabled={true}>
                                    <ActivityIndicator size="large" color="#ffffff"/>
                                </StyledButton>
                            )}

                            <Line />
                            <StyledButton testID='wordpress-button' wordpress={true} onPress={handleSubmit}>
                                <Fontisto name="wordpress" color={textInputBackground} size={25} />
                                <ButtonText wordpress={true}>Sign in with Wordpress</ButtonText>
                            </StyledButton>
                            <ExtraView>
                                <ExtraText>Don't have an account? </ExtraText>
                                <TextLink onPress={() => navigation.navigate("SignUp")}>
                                    <TextLinkContent>Signup</TextLinkContent>
                                </TextLink>
                            </ExtraView>

                            <ExtraView>
                                <TextLink onPress={() => navigation.navigate("Main Menu")}>
                                    <TextLinkContent>WHILE LOGIN IS FIXED</TextLinkContent>
                                </TextLink>
                            </ExtraView>
                                    
                        </StyledForm>)}
                    </Formik>
                </InnerContainer>
            </StyledContainer>
        </KeyboardAvoidWrap>
    );
}

const TextInput = ({
  label,
  icon,
  isPassword,
  hidePassword,
  setHidePassword,
  ...props
}) => {
  return (
    <View>
      <LabelIcon>
        <Octicons name={icon} size={30} color={company} />
      </LabelIcon>
      <StyleInputLabel>{label}</StyleInputLabel>
      <StyleTextInput {...props} />
      {isPassword && (
        <EyeIcon onPress={() => setHidePassword(!hidePassword)}>
          <Ionicons
            name={hidePassword ? "md-eye-off" : "md-eye"}
            size={30}
            color={placeholder}
          />
        </EyeIcon>
      )}
    </View>
  );
};

export default Login;
