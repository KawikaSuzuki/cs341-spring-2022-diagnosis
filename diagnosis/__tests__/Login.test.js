import React from 'react';
import renderer from 'react-test-renderer';

import Login from '../screens/Login';

describe('<Login />', () => {
    it("has 1 children", async () => {
        const tree = renderer.create(<Login />).toJSON();
        await expect(tree.children.length).toBe(1);
    });
    it('renders correctly', async () => {
        const tree = renderer.create(<Login />).toJSON();
        await expect(tree).toMatchSnapshot();
    });
});