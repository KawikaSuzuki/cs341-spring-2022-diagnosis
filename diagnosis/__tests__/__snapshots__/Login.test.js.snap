// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        className="sc-bdvvtL zxezy"
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          className="sc-gsDKAQ rZFaE"
        >
          <Image
            className="sc-dkPtRN fQWDve"
            resizeMode="contain"
            source={1}
          />
          <Text
            className="sc-hKwDye fAcRlN"
          >
            Account Login
          </Text>
          <View
            className="sc-eCImPb kMRaWM"
          >
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Username
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="username"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Password
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="********"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                secureTextEntry={true}
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <Text>
              

            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="sc-jrQzAO dafyie"
              >
                Login
              </Text>
            </View>
            <View
              className="sc-kDTinF glXNyz"
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text />
              <Text
                className="sc-jrQzAO gGIeVH"
                wordpress={true}
              >
                Sign in with Wordpress
              </Text>
            </View>
            <View
              className="sc-iqseJM DBbKg"
            >
              <Text
                className="sc-crHmcD bLjgwf"
              >
                Don't have an account? 
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="sc-bqiRlB kAwtfN"
                >
                  Signup
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
